### PR TITLE
DEVPROD-7426: Use Vitest in Spruce

### PR DESCRIPTION
**Pull request by sophstad**
DEVPROD-7426

### Description

- Replace instances of `jest.` with `vi.` (this is the reason for so many files changed!)
- Update snapshot testing so that each component within a story file gets its own snapshot file. This allows us to use Vitest's [toMatchFileSnapshot](https://vitest.dev/api/expect.html#tomatchfilesnapshot) function instead of the jest-specific-snapshot package.
- Explicitly install `jsdom` instead of using `jest-environment-jsdom`. I think this caused some snapshot updates in Parsley since the version got bumped.
- Implemented in Parsley here: https://github.com/evergreen-ci/ui/pull/122

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,11 +18,13 @@
     "test:jest": "jest --watchAll=false"
   },
   "resolutions": {
+    "@testing-library/dom": "10.1.0",
```
Comment (sophstad) [inline package.json:21]: Incorrect resolution of `@testing-library/dom` results in a lot of those extraneous `act()` warnings. Previously, additional versions were being installed by Storybook and LeafyGreen. Idea from [here](https://stackoverflow.com/a/76089260).

File: apps/parsley/src/components/LogRow/BaseRow/Highlighter.test.tsx [hunk 2]
```diff
@@ -8,7 +8,9 @@ describe("highlighter", () => {
         Test blah test blah
       </Highlighter>,
     );
-    expect(screen.getByDataCy("test-row")).toHaveStyle(`color: salmon`);
+    expect(screen.getByDataCy("test-row")).toHaveStyle(
+      `color: rgb(250, 128, 114)`,
```
[the remark this answers is not in the document]
Comment (sophstad) [reply apps/parsley/src/components/LogRow/BaseRow/Highlighter.test.tsx:12]: Yep!

File: apps/spruce/src/pages/taskHistory/ColumnHeaders/ColumnHeaders.test.tsx [hunk 3]
```diff
@@ -31,7 +31,9 @@ describe("columnHeaders (Task History)", () => {
       path: "/task-history/:projectId/:taskName",
       wrapper: ProviderWrapper,
     });
-    expect(screen.queryAllByDataCy("loading-header-cell")).toHaveLength(7);
+    await waitFor(() => {
```
[the remark this answers is not in the document]
Comment (sophstad) [reply apps/spruce/src/pages/taskHistory/ColumnHeaders/ColumnHeaders.test.tsx:34]: All of these tests are tests that failed when running with Vitest. My impression is these components include a `useQuery` call or other fetch that requires waiting for that promise to resolve. I am not too sure why the frameworks differ in this though 🤔

[the remark this answers is not in the document]
Comment (khelif96) [reply apps/spruce/src/pages/taskHistory/ColumnHeaders/ColumnHeaders.test.tsx:34]: Interesting that makes sense I wonder if treats the promises differently but it seems correct overall! Great work!

**Pull request merged**